### PR TITLE
feat: health items

### PR DESCRIPTION
**Pull request by cazala**
This PR adds the following:

### Components:

- `CounterBar`: It renders a bar using the value of the Counter component in the same entity. It takes a max value, a primary and a secondary color.

### Actions:

- `DAMAGE`: It triggers all the `on_damage` triggers nearby, taking the `radius` and `layer` params to filter out targets, and using the `hits` param as a multiplier
- `MOVE_PLAYER_HERE`: Moves the player to the entity's position
- `PLACE_ON_PLAYER`: Moves the entity to the player's position
- `ROTATE_AS_PLAYER`: Rotates the entity using the player's rotation
- `PLACE_ON_CAMERA`: Moves the entity to the camera's position
- `ROTATE_AS_CAMERA`: Rotates the entity using the camera's rotation
- `SET_POSITION`: Sets the X/Y/Z values of the position of the entity
- `SET_ROTATION`: Sets the X/Y/Z values of the rotation of the entity
- `SET_SCALE`: Sets the X/Y/Z values of the scale of the entity
- `FOLLOW_PLAYER`: Makes the entity start following the player. User can set `speed`, `minDistance` and lock the X/Y/Z axes.
- `STOP_FOLLOWING_PLAYER`: Makes the entity stop following the player
- `RANDOM`: It takes an array of actions and it executes only 1 randomly.
- `BATCH`: It takes an array of actions and it executes them all.
- `HEAL_PLAYER`: It will trigger the `ON_HEAL_PLAYER` trigger on any entity as long as it's a child of the PlayerEntity

### Triggers

- `ON_DAMAGE`: Triggered by a nearby `DAMAGE` action, the number of times configured in the `hits` param of the `DAMAGE` action
- `ON_GLOBAL_CLICK`: Click anywhere
- `ON_GLOBAL_PRIMARY`: Press E
- `ON_GLOBAL_SECONDARY`: Press F
- `ON_TICK`: On every frame
- `ON_HEAL_PLAYER`: Triggered by the `HEAL_PLAYER` action only if this entity is child of the PlayerEntity

### Trigger Conditions
- `WHEN_DISTANCE_TO_PLAYER_IS_LESS_THAN`: The condition will pass when the distance to the player position is less than the value set in the condition. It can be used on any trigger type.
- `WHEN_DISTANCE_TO_PLAYER_IS_LESS_THAN`: The condition will pass when the distance to the player position is more than the value set in the condition. It can be used on any trigger type.

### CODE REVIEW

**Review by cyaiox — APPROVED** [review]

File: src/actions.ts [hunk 1]
```diff
@@ -915,4 +1011,212 @@ export function createActionsSystem(
...
+  }
+
+  // SET_POSITION
+  function handleSetPosition(
+    entity: Entity,
+    payload: ActionPayload<ActionType.SET_POSITION>,
+  ) {
+    const transform = Transform.getMutableOrNull(entity)
+    if (transform) {
+      transform.position = Vector3.create(payload.x, payload.y, payload.z)
```
Comment (cyaiox) [inline src/actions.ts:1121]: should this position be relative to the scene, or is it okay to be global?

```
transform.position = Vector3.create(
  transform.position.x + payload.x,
  transform.position.y + payload.y,
  transform.position.z + payload.z
)
```

Comment (cazala) [reply src/actions.ts:1121]: mm i can see use cases for both, although you can do the relative use case by nesting a child entity and using the SET_POSITION action on the child. But I think I could add a `relative?: boolean` param and if present add the values to the current ones as in your example

Comment (cazala) [reply src/actions.ts]: done ✅

**Review by cyaiox — APPROVED** [review]
🚀 🚀

**Pull request merged**